core/server/api_container/server/startosis_engine/kurtosis_instruction/add_service/add_services.go

```diff
@@ -123,25 +124,48 @@ func (builtin *AddServicesCapabilities) Execute(ctx context.Context, _ *builtin_
 		renderedServiceConfigs[renderedServiceName] = renderedServiceConfig
 	}
 
-	startedServices, failedServices, err := builtin.serviceNetwork.AddServices(ctx, renderedServiceConfigs, parallelism)
+	serviceToUpdate := map[service.ServiceName]*service.ServiceConfig{}
+	serviceToCreate := map[service.ServiceName]*service.ServiceConfig{}
+	for serviceName, serviceConfig := range renderedServiceConfigs {
+		if _, found := builtin.serviceNetwork.GetServiceRegistration(serviceName); found {
+			serviceToUpdate[serviceName] = serviceConfig
+		} else {
+			serviceToCreate[serviceName] = serviceConfig
+		}
+	}
+
+	updatedServices, failedToBeUpdatedServices, err := builtin.serviceNetwork.UpdateServices(ctx, serviceToUpdate, parallelism)
+	if err != nil {
+		return "", stacktrace.Propagate(err, "Unexpected error occurred updating a batch of services")
+	}
+
+	startedServices, failedToBeStartedServices, err := builtin.serviceNetwork.AddServices(ctx, renderedServiceConfigs, parallelism)
```

Comment:
is it ok to add all the services?

Reply:
Nope deff not, that should be `serviceToCreate` here